instrumentation/spring-rabbit/src/main/java/brave/spring/rabbit/SpringRabbitTracing.java

```diff
@@ -60,6 +68,36 @@ public RabbitTemplate newRabbitTemplate(ConnectionFactory connectionFactory) {
     return rabbitTemplate;
   }
 
+  /** Instruments an existing rabbit template. */
+  public RabbitTemplate fromRabbitTemplate(RabbitTemplate rabbitTemplate) {
+    try {
+      MessagePostProcessor[] processorsArray = currentProcessorsWithTracing(
+              rabbitTemplate);
+      rabbitTemplate.setBeforePublishPostProcessors(processorsArray);
+      return rabbitTemplate;
+    }
+    catch (NoSuchFieldException | IllegalAccessException e) {
+      return rabbitTemplate;
+    }
+  }
+
+  private MessagePostProcessor[] currentProcessorsWithTracing(
+          RabbitTemplate rabbitTemplate)
+          throws NoSuchFieldException, IllegalAccessException {
+    Field field = RabbitTemplate.class
+            .getDeclaredField("beforePublishPostProcessors");
+    Collection<MessagePostProcessor> processors =
+            (Collection<MessagePostProcessor>) field.get(rabbitTemplate);
+    List<MessagePostProcessor> newProcessors = new ArrayList<>();
+    if (!processors.contains(tracingMessagePostProcessor)) {
```

Comment:
not sure this will work as we don't do any deep equals etc. Maybe we should just look for an instance of?